### PR TITLE
Fix bazel build

### PR DESCRIPTION
**Pull request by camdencheek**
The new gRPC stuff cannot be used in sourcegraph because the bazel build fails.

This updates the build files (with some pain). Both `go test ./...` and `bazel test //...` now pass.

Side note: should we add bazel build checks to CI?

### CODE REVIEW

File: grpc/v1/query.proto [hunk 1]
```diff
@@ -2,10 +2,6 @@ syntax = "proto3";
 
 package grpc.v1;
 
-import "google/protobuf/duration.proto";
-import "google/protobuf/empty.proto";
-import "google/protobuf/timestamp.proto";
-
```
Comment (camdencheek) [inline grpc/v1/query.proto:8]: Bazel nicely told me these weren't used

File: BUILD.bazel [hunk 2]
```diff
@@ -117,9 +117,9 @@ go_test(
         "read_test.go",
         "tombstones_test.go",
     ],
-    data = [":testdata"],
+    data = ["//testdata"],
     embed = [":zoekt"],
-    embedsrcs = ["//testdata:search_result_1.pb"],  #keep
+    embedsrcs = ["//testdata:testdata"],  #keep
```
Comment (camdencheek) [inline BUILD.bazel:122]: This was the painful part. Gazelle doesn't seem to be able to figure out how this embed works.

File: grpc/v1/BUILD.bazel [hunk 3]
```diff
@@ -4,11 +4,14 @@ load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 
 proto_library(
     name = "v1_proto",
-    srcs = ["webserver.proto"],
+    srcs = [
+        "query.proto",
+        "webserver.proto",
+    ],
+    strip_import_prefix = "/grpc/v1",
```
Comment (camdencheek) [inline grpc/v1/BUILD.bazel:11]: This was also a new one for me. Apparently this is needed to make proto imports work for both `buf` and `bazel`.

**Review by keegancsmith — APPROVED** [review]
> Side note: should we add bazel build checks to CI?

yes, lets prevent future pain.

**Pull request merged**